src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -20,6 +20,15 @@ namespace System.Management.Automation.Runspaces
     /// </summary>
     internal sealed class LocalPipeline : PipelineBase
     {
+        // Default stack size in bytes for local pipeline threads.
+        //
+        // Each platform uses different default for the thread stack size:
+        //      - Windows 2 MB
+        //      - Linux   8 Mb
+        //      - MacOs   512 KB
+        // We should use the same default for all platforms to get predictable behavior.
+        internal const int DefaultPipelineStackSize = 10_000_000;
```

Comment:
It may be worthwhile to add a comment that the 10MB default is inherited from Windows PowerShell